pkg/limitador/k8s_objects.go

```diff
@@ -0,0 +1,141 @@
+package limitador
+
+import (
+	limitadorv1alpha1 "github.com/3scale/limitador-operator/api/v1alpha1"
+	appsv1 "k8s.io/api/apps/v1"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+)
+
+const (
+	DefaultVersion   = "latest"
+	ServiceName      = "limitador"
+	ServiceNamespace = "default"
+	Image            = "quay.io/3scale/limitador"
+	StatusEndpoint   = "/status"
+)
+
+func LimitadorService() *v1.Service {
+	return &v1.Service{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Service",
+			APIVersion: "v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      ServiceName,
+			Namespace: ServiceNamespace,
+			Labels:    labels(),
+		},
+		Spec: v1.ServiceSpec{
+			Ports: []v1.ServicePort{
+				{
+					Name:       "http",
+					Protocol:   v1.ProtocolTCP,
+					Port:       8080,
+					TargetPort: intstr.FromString("http"),
+				},
+				{
+					Name:       "grpc",
+					Protocol:   v1.ProtocolTCP,
+					Port:       8081,
+					TargetPort: intstr.FromString("grpc"),
+				},
+			},
+			Selector:  labels(),
+			ClusterIP: v1.ClusterIPNone,
+			Type:      v1.ServiceTypeClusterIP,
+		},
+	}
+}
+
+func LimitadorDeployment(limitador *limitadorv1alpha1.Limitador) *appsv1.Deployment {
+	replicas := int32(limitador.Spec.Replicas)
```

Comment:
With the current implementation, if the user does not specify the Replicas field in the CR this value will be the zero-value of int32 which is 0. See my https://github.com/3scale-labs/limitador-operator/pull/2/files#r553854803 comment for some insight on this

Reply:
You're right. Fixed :+1: